### PR TITLE
Move all GUI code under a single folder and hook up robot status table to robot diagnostics

### LINKED ISSUE #1304: Move all GUI code under a single shared folder
### Description of the task

Currently code for the Visualizer and Robot Diagnostics are in completely separate folders. We should consolidate our GUI code under a single folder to facilitate sharing GUI components and code.

### Acceptance criteria

- [ ] Move the Visualizer and Robot Diagnostics into a new shared GUI folder

### LINKED ISSUE #1306: Create a generic widget for status messages
### Description of the task

The Visualizer has a widget for displaying robot status and error messages. This should be made generic so it can be shared by the Robot Diagnostics GUI.

### Acceptance criteria

- [ ] Make the RobotStatus widget generic / shareable.

### PR DESCRIPTION
**Pull request by LiCody**
### Description

Restructured the Visualizer and RobotDiagnostics to be under a single ```qt_gui``` folder with generic widgets moved to a shared folder. The robot status widget was moved into the shared folder and also hooked up to robot diagnostics 

### Testing Done

Visual inspection of robot diagnosis and visualizer, CI

### Resolved Issues

resolves #1304 
resolves #1306 

### Review Checklist

**_It is the reviewers responsibility to also make sure every item here has been covered_**

- [x] **Function & Class comments**: All function definitions (usually in the `.h` file) should have a javadoc style comment at the start of them. For examples, see the functions defined in `thunderbots/software/geom`. Similarly, all classes should have an associated Javadoc comment explaining the purpose of the class.
- [x] **Remove all commented out code**
- [x] **Remove extra print statements**: for example, those just used for testing
- [x] **Resolve all TODO's**: All `TODO` (or similar) statements should either be completed or associated with a github issue
- [x] **Justify drops in code coverage**: If this PR results in a non-trivial drop in code coverage (a bot should post a coverage diagram as a comment), please justify why we can't test the code that's not covered.

### CODE REVIEW

**Review by jonathanlew — CHANGES_REQUESTED** [review]
Good work! Initial comment: Can we move generic_widgets/robot_status* to generic_widgets/robot_status/robot_status*?

**Review by MathewMacDougall — CHANGES_REQUESTED** [review]
Looks good! And good catch changing the `.ui` files. My only suggestion is to rename the `qt_gui` folder to just `gui` so it's not named framework-specific (so we don't need to rename it in the future ideally)

**Review by akhilveeraghanta — COMMENTED** [review]
lgtm :tada: , pending other comments and merge conflict

File: src/software/gui/visualizer/drawing/obstacle_artist.h [hunk 1]
```diff
@@ -1,13 +1,13 @@
 #pragma once
 
-#include <QtWidgets/QGraphicsScene>
+#include "QtWidgets/QGraphicsScene"
```
Comment (jonathanlew) [inline src/software/gui/visualizer/drawing/obstacle_artist.h]: Curious about why we're making this change

Comment (LiCody) [reply src/software/gui/visualizer/drawing/obstacle_artist.h]: oops that is from a bazel error, fixed now

**Review by jonathanlew — APPROVED** [review]
LGTM! :tada:

**Review by MathewMacDougall — APPROVED** [review]
LGTM :+1 nice work :tada:

**Review by garethellis0 — APPROVED** [review]
LGTM :+1: :tada:

**Pull request merged**